server_test.go

```diff
@@ -466,3 +468,12 @@ func startTargetServer() *httptest.Server {
 	testServer := httptest.NewServer(instance)
 	return testServer
 }
+
+func newServer(t *testing.T, target string, opts ...ServerOption) *Server {
+	server, err := NewServer(target, opts...)
+	require.NoError(t, err)
+
+	server.responseHandler = server.createResponseHandler()
```

Comment:
Would it make sense to define `ServeHTTP` on `server` that could ensure `createResponseHandler` and `CreateHandler()` are called before we start serving requests?